float32[]   params
float32[]   prob
